File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.py
from collections import Counter

class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        n = len(nums)
        freq = Counter(nums)

        if k == 1:
            ans = -1
            for num, count in freq.items():
                if count == 1:
                    ans = max(ans, num)
            return ans

        if k == n:
            return max(nums)

        candidates = []
        if freq[nums[0]] == 1:
            candidates.append(nums[0])

        if freq[nums[-1]] == 1:
            candidates.append(nums[-1])

        return max(candidates) if candidates else -1

File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.cpp
#include<vector>
#include<unordered_map>
#include<algorithm>

class Solution {
public:
    int largestInteger(std::vector<int>& nums, int k) {
        int n = nums.size();
        std::unordered_map<int, int> freq;
        for(int num : nums) {
            freq[num]++;
        }

        if(k==1) {
            int ans = -1;
            for(const auto& [num, count] : freq) {
                if(count == 1) {
                    ans = std::max(ans, num);
                }
            }
            return ans;
        }

        if(k == n) {
            return *std::max_element(nums.begin(), nums.end());
        }

        int ans = -1;
        if(freq[nums[0]] == 1) {
            ans = std::max(ans, nums[0]);
        }
        if(freq[nums[n - 1]] == 1) {
            ans = std::max(ans, nums[n - 1]);
        }
        return ans;
    }
};